src/Form/matching.js

```diff
@@ -160,17 +160,19 @@ class Matching {
      * @param {HTMLInputElement|HTMLSelectElement} input
      * @param {HTMLFormElement} formEl
      * @param {{isLogin?: boolean}} [opts]
-     * @returns {SupportedSubTypes | string}
+     * @returns {SupportedTypes}
      */
     inferInputType (input, formEl, opts = {}) {
         const presetType = input.getAttribute(ATTR_INPUT_TYPE)
+        // @ts-ignore
```

Comment:
This specifically fails because it's any string, we could switch/case statement check the properties too but with the current setup it would be more duplication of the type checks.